node/src/blockchain/blockchain_interface.rs

```diff
@@ -2331,18 +2337,20 @@ mod tests {
             BlockchainError::QueryFailed(
                 "Don't query so often, it gives me a headache".to_string(),
             ),
+            BlockchainError::UninitializedBlockchainInterface,
         ];
         let pretty_print_closure = |err_to_resolve: &BlockchainError| match err_to_resolve {
             BlockchainError::InvalidUrl => (err_to_resolve.to_string(), 11),
```

Comment:
It's not clear to me that these check numbers are necessary. I don't object to additional assertions, but if they're there they should be simple and easy to understand.

You can take a look at the `enum_iterator` crate, but I'm thinking it may be a case where the juice isn't worth the squeeze.

https://docs.rs/enum-iterator/latest/enum_iterator